### DISCUSSION TITLE
reaction products with single bond connecting aromatic atoms generated and cannot be sanitised

### DISCUSSION THREAD

**Discussion by cottonkiet** [rdkit/rdkit#5138] [Q&A]
Hi

I am using the following reaction:
`scheme = '[#7:7]-[c:1]1[c:2][c:3][c:4]([c:5][c:6]1)[S:8](=O)(=O)[#7:9]-[c:10]1[#6,#7,#8,#16;a:14][#6,#7,#8,#16;a:13][#6,#7,#8,#16;a:12][#6,#7,#8,#16;a:11]1>>[#7:7]-[#6:1]-1=[#6:2]-[#6:3]=[#6:4](-[#6:5]=[#6:6]-1)[S:8]([#7:9])(=O)=O.[c:10]1[#6,#7,#8,#16;a:14][#6,#7,#8,#16;a:13][#6,#7,#8,#16;a:12][#6,#7,#8,#16;a:11]1'`

with one reactant:
`react = Chem.MolFromSmiles('Cc1cc(NS(=O)(=O)c2ccc(N)cc2)no1')`
But the second product I produce is not correct
Cc1-c-c-o-n-1
I tried to sanitise the molecule, only to get the following:
CC1CCON1

In theory, it should produce a product
Cc1ccon1

My input molecule is aromatic, and smarts is aromatic as far as the five membered ring is concerned. Why will I get the single bonds between the aromatic atoms? How should I avoid the error when doing the reaction or correctly sanitise the product smiles? Thanks so much for your time! 

To reproduce

```
from rdkit import Chem
from rdkit.Chem import AllChem
react = Chem.MolFromSmiles('Cc1cc(NS(=O)(=O)c2ccc(N)cc2)no1')
display(react)
scheme = '[#7:7]-[c:1]1[c:2][c:3][c:4]([c:5][c:6]1)[S:8](=O)(=O)[#7:9]-[c:10]1[#6,#7,#8,#16;a:14][#6,#7,#8,#16;a:13][#6,#7,#8,#16;a:12][#6,#7,#8,#16;a:11]1>>[#7:7]-[#6:1]-1=[#6:2]-[#6:3]=[#6:4](-[#6:5]=[#6:6]-1)[S:8]([#7:9])(=O)=O.[c:10]1[#6,#7,#8,#16;a:14][#6,#7,#8,#16;a:13][#6,#7,#8,#16;a:12][#6,#7,#8,#16;a:11]1'
rxn = AllChem.ReactionFromSmarts(scheme)
display(rxn)
prod = rxn.RunReactants([react])
len(prod)
display(prod[0][1])
Chem.SanitizeMol(prod[0][1])
display(prod[0][1])
```

Configuration (please complete the following information):

RDKit version: 2021.09.5
OS: Windows
Python version (if relevant): 3.9
Are you using conda? Yes
If you are using conda, which channel did you install the rdkit from? default

**Comment by greglandrum** [comment]
I think it's fair to expect the RDKit to get this right since you haven't changed anything from reactants to products, but since it's not doing that the easiest workaround is to just specify that the bonds in the products are aromatic (an unspecified bond in SMARTS is "Single or Aromatic") like this:
```
scheme = '[#7:7]-[c:1]1[c:2][c:3][c:4]([c:5][c:6]1)[S:8](=O)(=O)[#7:9]-[c:10]1[#6,#7,#8,#16;a:14][#6,#7,#8,#16;a:13][#6,#7,#8,#16;a:12][#6,#7,#8,#16;a:11]1>>[#7:7]-[#6:1]-1=[#6:2]-[#6:3]=[#6:4](-[#6:5]=[#6:6]-1)[S:8]([#7:9])(=O)=O.[c:10]:1:[*:14]:[*:13]:[*:12]:[*:11]:1'
```
Notice that I've also replaced the queries you had in the products with simple any queries  (which will be interpreted by the RDKit as "don't make any changes to the atom"). There's no reason to have the complex queries in the products since they wouldn't be used for anything and just make the reaction harder to read.

**Comment by greglandrum** [comment]
An alternate solution, if you aren't sure the bonds will be aromatic (I know that in this case you are), would be to use "Any" bonds in the products:
```
scheme = '[#7:7]-[c:1]1[c:2][c:3][c:4]([c:5][c:6]1)[S:8](=O)(=O)[#7:9]-[c:10]1[#6,#7,#8,#16;a:14][#6,#7,#8,#16;a:13][#6,#7,#8,#16;a:12][#6,#7,#8,#16;a:11]1>>[#7:7]-[#6:1]-1=[#6:2]-[#6:3]=[#6:4](-[#6:5]=[#6:6]-1)[S:8]([#7:9])(=O)=O.[c:10]~1~[*:14]~[*:13]~[*:12]~[*:11]~1'
```
This will just copy the corresponding bond order from the reactant molecule into the products.
There is some documentation about this available: https://rdkit.org/docs/RDKit_Book.html#some-features

**Reply by cottonkiet** [reply]
Thanks so much! 
Both solution works. However, I am getting some error message like this, although I can run the reaction correctly:
RDKit ERROR: Pre-condition Violation
RDKit ERROR: getExplicitValence() called without call to calcExplicitValence()
RDKit ERROR: Violation occurred on line 194 in file D:\bld\rdkit_1646569338444\work\Code\GraphMol\Atom.cpp
RDKit ERROR: Failed Expression: d_explicitValence > -1
Does this indicate there is something wrong with my smarts?

**Reply by greglandrum** [reply]
I can't help you with this if you don't also tell us which line of your python code generated these errors

**Reply by cottonkiet** [reply]
It does not show which line of my python code generated these errors.
![image](https://user-images.githubusercontent.com/102205886/160410696-350bc7df-e0b0-4658-bc9d-213237b81388.png)
